modules/rest-compatibility/src/main/java/org/elasticsearch/rest/compat/RestCompatPlugin.java

```diff
@@ -51,7 +51,7 @@
         Supplier<DiscoveryNodes> nodesInCluster
     ) {
         if (Version.CURRENT.major == 8) {
-            return List.of(
+            return validatedList(7,
```

Comment:
+1 to adding validation... I would have a minor preference to have a static validateCompatibleHandlers(Collection<RestHandler> handlers) that is located in a more central location and that assumes Version.CURRENT.major - 1 to help make this more generic. 

Also, I think normal IllegalStateException or the like may be preferable to the assert here.

Reply:
agree - IllegalStateException fits better here.
Also agree that this could be moved to some more generic place once we have more compat rest handlers in other modules. For the time being I would prefer to keep this in that class (as I don't know the right place since there are no more compat handlers in other modules yet)